Kinvey.Tests/DataStoreTests/DataStoreAutoIntegrationTests.cs

```diff
@@ -6982,6 +7650,301 @@ public async Task TestPushRecreatedDataNetworkConnectionAvailableAsync()
+        {
+            // Setup
+            kinveyClient = BuildClient("5");
+
+            if (MockData)
+            {
+                MockResponses(11);
+            }
+
+            // Arrange
+            var user = await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
+
+            var autoToDoStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.AUTO, kinveyClient);
+            var syncToDoStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.SYNC, kinveyClient);
+            var networkToDoStore = DataStore<ToDo>.Collection(toDosCollection, DataStoreType.NETWORK, kinveyClient);
+
+            var toDos = new List<ToDo>
+            {
+               
+            };
+
+            var toDo1 = new ToDo { Name = "Name1", Details = "Details1", Value = 1 };
+            toDos.Add(toDo1);
+
+            var toDo2 = new ToDo { Name = "Name2", Details = "Details2", Value = 2 };
+            toDo2 = await autoToDoStore.SaveAsync(toDo2);
+            toDo2.Name = "Name22";
+            toDo2.Details = "Details22";
+            toDo2.Value = 22;
+            toDos.Add(toDo2);
+
+            var toDo3 = new ToDo { Name = "Name3", Details = "Details3", Value = 3 };
+            toDos.Add(toDo3);
+
+            var toDo4 = new ToDo { ID = Guid.NewGuid().ToString(), Name = "Name4", Details = "Details4", Value = 4 };
+            toDos.Add(toDo4);
+
+            // Act
+            SetRootUrlToKinveyClient(unreachableUrl);
+            var savedToDos = await autoToDoStore.SaveAsync(toDos);
+            SetRootUrlToKinveyClient(kinveyUrl);
+
+            var pushResponse = await autoToDoStore.PushAsync();
+
+            var pendingWriteActions = kinveyClient.CacheManager.GetSyncQueue(toDosCollection).GetAll();
+            var existingToDosNetwork = await networkToDoStore.FindAsync();
+
+            // Teardown
+            await networkToDoStore.RemoveAsync(pushResponse.PushEntities[0].ID);
+            await networkToDoStore.RemoveAsync(pushResponse.PushEntities[1].ID);
+            await networkToDoStore.RemoveAsync(pushResponse.PushEntities[2].ID);
+            await networkToDoStore.RemoveAsync(pushResponse.PushEntities[3].ID);
+
+            // Assert
+            Assert.AreEqual(4, pushResponse.PushCount);
+            Assert.IsNotNull(pushResponse.PushEntities.FirstOrDefault(e => e.Name.Equals(toDos[0].Name) && e.Details.Equals(toDos[0].Details) && e.Value == toDos[0].Value));
+            Assert.IsNotNull(pushResponse.PushEntities.FirstOrDefault(e => e.Name.Equals(toDos[1].Name) && e.Details.Equals(toDos[1].Details) && e.Value == toDos[1].Value));
+            Assert.IsNotNull(pushResponse.PushEntities.FirstOrDefault(e => e.Name.Equals(toDos[2].Name) && e.Details.Equals(toDos[2].Details) && e.Value == toDos[2].Value));
+            Assert.IsNotNull(pushResponse.PushEntities.FirstOrDefault(e => e.Name.Equals(toDos[3].Name) && e.Details.Equals(toDos[3].Details) && e.Value == toDos[3].Value));
+
```

Comment:
We should check the _id for items 1 and 3

Reply:
Ok